File: docs/snippets/all/archetypes/mcap_schema_simple.cpp
// Log a simple MCAP schema definition.

#include <rerun.hpp>
#include <string>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_mcap_schema");
    rec.spawn().exit_on_failure();

    // Example ROS2 message definition for a simple Point message
    const std::string point_schema = "float64 x\nfloat64 y\nfloat64 z";

    rec.log(
        "mcap/schemas/geometry_point",
        rerun::archetypes::McapSchema(
            42,
            "geometry_msgs/msg/Point",
            "ros2msg",
            rerun::components::Blob(point_schema)
        )
    );
}

File: docs/snippets/all/archetypes/mcap_schema_simple.rs
//! Log a simple MCAP schema definition.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_mcap_schema").spawn()?;

    // Example ROS2 message definition for a simple Point message
    let point_schema = "float64 x\nfloat64 y\nfloat64 z";

    rec.log(
        "mcap/schemas/geometry_point",
        &rerun::McapSchema::new(
            42,
            "geometry_msgs/msg/Point",
            "ros2msg",
            point_schema.as_bytes(),
        ),
    )?;

    Ok(())
}
